packages/orb-sync-lib/src/invoice-utils.ts

```diff
@@ -0,0 +1,32 @@
+import { Invoice } from 'orb-billing/resources';
+
+const PLAN_LINE_ITEM_NAME_ENDS_IN = 'Plan';
+
+/**
+ * Returns the billing cycle the given invoice's plan line item applies to.
+ * If no plan line item is present, null is returned.
+ */
+export function getBillingCycleFromInvoice(invoice: Invoice): { start: string; end: string } | null {
+  const planLineItem = findPlanLineItem(invoice.line_items);
+
+  // No plan line item found.
+  // Is the case for e.g. invoices that include usage line items for the past billing cycle only or
+  // invoices that include a fixed price line item other than the plan line item only
+  if (!planLineItem) {
+    return null;
+  }
+
+  return {
+    start: planLineItem.start_date,
+    end: planLineItem.end_date,
+  };
+}
+
+function findPlanLineItem(lineItems: Invoice.LineItem[]): Invoice.LineItem | undefined {
+  return lineItems.filter(
+    (item) =>
+      item.price?.price_type === 'fixed_price' &&
+      item.price.billable_metric === null &&
+      item.name.endsWith(PLAN_LINE_ITEM_NAME_ENDS_IN)
```

Comment:
The same invoice can never contain two plan items can it?

Reply:
Actually the check on the name is enough. The other two conditions are more of an additional safety net. Since there's no official concept of a "plan item" in Orb, every fixed price item actually looks like what we consider a plan item.
In case of HIPAA for example, the moment we add HIPAA to a subscription, there would be an invoice with one line item only. If we didn't check the name we would think that this is a plan item. The invoice for the next billing cycle after HIPAA was added would include two line items with a "fixed price" and no billable metric (1 for e.g. Team Plan, 1 for HIPAA).